nats/aio/client.py

```diff
@@ -768,6 +772,28 @@ def _process_connect_init(self):
         self._server_info = json.loads(info.decode())
         self._max_payload = self._server_info["max_payload"]
 
+        if self._server_info['tls_required']:
+            ssl_context = self.options.get('tls')
+            if not ssl_context:
+                raise NatsError('no ssl context provided')
+
+            transport = self._io_writer.transport
+            sock = transport.get_extra_info('socket')
+            if not sock:
+                # This shouldn't happen
+                raise NatsError('unable to get socket')
+            yield from self._io_writer.drain()  # just in case something is left
+
+            self._io_reader, self._io_writer = \
+                yield from asyncio.open_connection(
+                    loop=self._loop,
+                    limit=DEFAULT_BUFFER_SIZE,
+                    sock=sock,
```

Comment:
Nice... I missed that calling `open_connection`  this way does not create another connection and works ok to wrap it using the TLS context.